### ISSUE TITLE
Ghost is invisible

### ISSUE THREAD

**Issue by regulus79** [Irevol/minetest-citadel#40]
When meeting the ghost at various places in the game, the entity has no texture or model.

When I rightclick it, it still gives the dialogue, but other than that it is not visible. I am guessing this may be a problem on my side, since Nathan.S's video review seemed to load the ghosts just fine.

![image](https://github.com/Irevol/minetest-citadel/assets/117475203/a27f01a5-ef21-493a-a428-5c8718b233ff)

**Comment by Warr1024** [comment]
Does it happen consistently for certain ghosts?  Is it random or is repeatable?  How often does it happen?

**Comment by regulus79** [comment]
It happens to every entity, even the crystal at the top of the castle (I didn't even know there was a crystal up there until I watched Nathan's video)

**Comment by Warr1024** [comment]
Can you post your minetest.conf file, especially anything related to graphics settings?

**Comment by regulus79** [comment]
Here is my minetest.conf file
```
menu_last_game = citadel
maintab_LAST = local
enable_fog = false
remote_port = 30002
screen_w = 1920
screen_h = 1011
name = Regulus
address = mt.cwiki.cyou
selected_world_path = /home/*****/.minetest/worlds/run103
mainmenu_last_selected_world = 173
mute_sound = true
fixed_map_seed = 863765
mg_name = singlenode
creative_mode = false
enable_damage = true
fast_move = false
free_move = false
noclip = true
continuous_forward = false
keymap_decrease_viewing_range_min = KEY_MINUS
keymap_autoforward = KEY_KEY_F
doubletap_jump = false
pitch_move = false
keymap_increase_viewing_range_min = KEY_PLUS
keymap_cmd_local = KEY_PERIOD
autojump = false
aux1_descends = false
server_announce = false
viewing_range = 50
world_config_selected_mod = 215
time_speed = 0
mgv6_spflags = mudflow,biomeblend,trees,nosnowbiomes,nojungles,noflat
enable_server = false
port = 30000
mgv7_spflags = caverns,floatlands,ridges,mountains
vehicles_nodes = true
mgfractal_spflags = noterrain
block_send_optimize_distance = 12
wieldview_update_time = 2
wieldview_node_tiles = false
vehicle_mash.api_mode = false
language = en
main_menu_technical_settings = true

enable_tnt = true

sound_volume = 0.629999995


enable_client_modding = true
csm_restriction_flags = 0
minimap_shape_round = true
update_last_checked = 1704126929
shadow_map_texture_32bit = true
shadow_map_max_distance = 62
shadow_map_texture_size = 512
shadow_levels = 1
shadow_map_color = false
shadow_filters = 0
enable_dynamic_shadows = false
update_last_known = 5008000


secure.trusted_mods = jitprofiler, mineysocket, ia_nodes

enable_shaders = false

sd_enable_devtools=true
secure.enable_security = true
secure.http_mods = llm
keymap_rangeselect = KEY_KEY_R
fullscreen = false
no_mtg_notification = true

```

**Comment by Warr1024** [comment]
I managed to repro it by just disabling shaders.  Looks like it's an engine bug ... one which I was pretty sure we had fixed a long time ago (like 5.6 or something) so I don't see why it should be back now...

**Comment by Warr1024** [comment]
Engine bug filed:

https://github.com/minetest/minetest/issues/14204

**Comment by andyca111100** [comment]
I guess there should be a warning for now like "you should enable shaders, or the ghost will be too realistic (invisible)"

**Comment by Warr1024** [comment]
It looks like there is a bug with using opacity <= 127 when shaders are not enabled. If you make sure that the opacity is >= 128 then it should work correctly in both cases.  Even if the bug is fixed for 5.9, until you are ready to drop support for 5.8 and earlier, you will probably need to do this at least as a workaround.  Hopefully the increase in opacity is not enough to make the ghost look "too solid".  If we need it to be more transparent, until the upstream bug is fixed, we would need to use some stippling/dithering style of transparency to make up the difference.

**Comment by Irevol** [comment]
I think the higher opacity looks ok. Thanks for figuring it out!
